### DISCUSSION TITLE
How to add hidden files on macOS

### DISCUSSION THREAD

**Discussion by d5ve** [borgbase/vorta#850] [FAQ]
**Describe the bug**
I can see hidden files in the "Add File" sources browser (by hitting `Cmd-shift-.`), however they are not selectable, and appeared greyed-out. This means I can't easily backup files like `.bashrc` etc.

I **can** add hidden directories like `.ssh` or `.vim` using the "Add Folder" sources browser.

Is there a way to add dotfiles to the backup sources? I'm happy to manually edit a config file if necessary, though I don't know where they are for Vorta.

**To Reproduce**
1. Launch 'Vorta for Borg Backup' from the macOS Status menu.
2. Click on 'Sources'
3. Click on 'Add File'
4. Hit `Cmd-shift-.` to have the file browser show hidden files.
5. See `.bashrc` in the list, but it's not able to be selected.

**Desktop (please complete the following information):**
 - OS: macOS Catalina
 - Vorta version 0.6.23
 - Borg version 1.1.10

**Comment by rbange** [comment]
@d5ve I experienced the same bug today, but was not able to quick-fix it in the code :/

Nevertheless, you can work around this by modifying `settings.db` in `~/Library/ApplicationSupport/Vorta/`. It is a sqlite db so get a browser and add a new entry to the table `sourcedirmodel`. 

Just make sure that Vorta is not running while doing this.

**Comment by ghost** [comment]
Hi - just tried Vorta for the first time and love it!

...but immediately saw this same issue.  Will try the workaround - thanks!

**Comment by m3nu** [comment]
Qt probably has an option to show hidden files.

Still not sure this is needed, since hidden files will be included in the backup, if you select the parent folder.

**Comment by ghost** [comment]
...I don't want to select the parent folder as that is my user folder containing too many things I do not wish to backup.

**Comment by m3nu** [comment]
In that case we would need a switch to show hidden files in the selector.

Would this work as checkbox in the *Misc* options?

**Comment by ghost** [comment]
that would be fine - thanks!

**Comment by m3nu** [comment]
I can confirm that it's possible to add hidden folders, but not hidden files. Let's see if there is an option for that.

**Comment by mwmcode** [comment]
Hello, 
Any updates on this one? 
I have the same scenario (I have a `.folder` that I want to backup and cannot include its parent since it's too big)

![Screen Shot 2020-12-12 at 3 46 51 pm](https://user-images.githubusercontent.com/13101565/101979807-5ef7a580-3cb4-11eb-842e-7b3b50f8708a.png)

**Comment by samuel-w** [comment]
As a workaround, we could override the native file dialog and replace it with the Qt dialog. https://doc.qt.io/qt-5/qfiledialog.html#Option-enum

Also, I found this: https://www.howtogeek.com/211496/how-to-hide-files-and-view-hidden-files-on-mac-os-x/. Seems to just be Command+Shift+Period to show files.

**Comment by mwmcode** [comment]
`Command + Shift + .` works (displays hidden folders/files) in the finder with _Adding File(s)_  option but not with _Adding Folder(s)_

Thanks @rbange the workaround worked.

**Reply by ghost** [reply]
Hi @m3nu , Have you selected this as the answer?  It does not allow you to select hidden files, only see them.

**Reply by m3nu** [reply]
Right. It allows to select hidden *folders*, but not files.

Still a good workaround, but doesn't answer the original question. That's true.

**Reply by ghost** [reply]
Why did you close the associated bug report?  ...are you not intending on fixing it anymore?

**Reply by m3nu** [reply]
I moved older issues that weren't related to things we can fix in Vorta to the discussion part. The connected issue is closed automatically and all comments are copied over.

I don't really know what could be changed to address the issue and lack the time to investigate. But if you can open a pull request to address the problem, I'll be happy to review it.

**Comment by ghost** [comment]
> Also, I found this: https://www.howtogeek.com/211496/how-to-hide-files-and-view-hidden-files-on-mac-os-x/. Seems to just be Command+Shift+Period to show files.

This does allow you to see `.files` and `.folders`, but it does not allow you to select the `.files`.

**Reply by alysson-souza** [reply]
A workaround is to `Ctrl+Click` the dot file you want to backup in Finder, press `Option` and then select the `Copy ".dotfile" as Pathname` menu item.  Now go to Vorta and click on the button `Paste Folders/files` button. You will then see that the dotfile was added as a source file.

**Reply by ghost** [reply]
Brilliant - thanks so much for that tip! 👍
